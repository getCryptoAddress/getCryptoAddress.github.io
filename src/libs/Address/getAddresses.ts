import { getAddress } from "micro-btc-signer";
import { NETWORK, TEST_NETWORK } from "micro-btc-signer";
export default function getAddresses(
  privateKey: Uint8Array,
  isTestnet = false
) {
  const cache: any = {};

  const network = isTestnet ? TEST_NETWORK : NETWORK;

  return Object.freeze({
    get tr() {
      if (!cache["tr"]) {
        cache["tr"] = getAddress("tr", privateKey, network);
      }
      return cache["tr"];
    },
    get pkh() {
      if (!cache["pkh"]) {
        cache["pkh"] = getAddress("pkh", privateKey, network);
      }
      return cache["pkh"];
    },
    get wpkh() {
      if (!cache["wpkh"]) {
        cache["wpkh"] = getAddress("wpkh", privateKey, network);
      }
      return cache["wpkh"];
    },
  });
}
