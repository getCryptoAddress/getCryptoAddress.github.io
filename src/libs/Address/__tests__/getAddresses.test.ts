import { describe, it, expect } from "vitest";
import getAddresses from "../getAddresses";

const key1 = new Uint8Array([
  232, 205, 117, 44, 180, 113, 245, 200, 213, 19, 97, 187, 70, 157, 86, 111, 19,
  131, 14, 18, 196, 57, 9, 203, 64, 196, 19, 26, 73, 35, 126, 80,
]);
const key2 = new Uint8Array([
  233, 205, 117, 44, 180, 113, 245, 200, 213, 19, 97, 187, 70, 157, 86, 111, 19,
  131, 14, 18, 196, 57, 9, 203, 64, 196, 19, 26, 73, 35, 126, 80,
]);
describe("[Address][Util] getAddresses", () => {
  describe("Main-net", () => {
    it("Should return correct addresses for key1", async () => {
      expect(JSON.stringify(getAddresses(key1))).toMatchSnapshot();
    });

    it("Should return correct addresses for key2", async () => {
      expect(JSON.stringify(getAddresses(key2))).toMatchSnapshot();
    });
  });

  describe("Test-net", () => {
    it("Should return correct test-net addresses for key1", async () => {
      expect(JSON.stringify(getAddresses(key1, true))).toMatchSnapshot();
    });

    it("Should return correct test-net addresses for key2", async () => {
      expect(JSON.stringify(getAddresses(key2, true))).toMatchSnapshot();
    });
  });
});
