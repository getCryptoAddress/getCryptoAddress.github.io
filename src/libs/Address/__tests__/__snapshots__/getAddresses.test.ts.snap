// Vitest Snapshot v1

exports[`[Address][Util] getAddresses > Main-net > Should return correct addresses for key1 1`] = `"{\\"tr\\":\\"bc1pdwq0u7qr2stgygm4ncqhyve4m06zm0czelvyhpwcesxsdj3hkqvq9me9wc\\",\\"pkh\\":\\"1E8VuVzEnUECHU1rHvhB8mabrxiNt5BN32\\",\\"wpkh\\":\\"bc1qjqz2el2le7zqj7jsns4ndcmqfnkgwux3rjhsdv\\"}"`;

exports[`[Address][Util] getAddresses > Main-net > Should return correct addresses for key2 1`] = `"{\\"tr\\":\\"bc1p2l8c6wap8c93726006v44sdmmv566mvrez7483pda0ylrg6fnaasuj6uzu\\",\\"pkh\\":\\"1NUmDDtSvxiRpjouKP9Ft99ZhSw1pqDCCn\\",\\"wpkh\\":\\"bc1qawdvw9jjf6jql4y7gtv8pfrq3zhmemx9cegmwr\\"}"`;

exports[`[Address][Util] getAddresses > Test-net > Should return correct test-net addresses for key1 1`] = `"{\\"tr\\":\\"tb1pdwq0u7qr2stgygm4ncqhyve4m06zm0czelvyhpwcesxsdj3hkqvqjn025h\\",\\"pkh\\":\\"mteTCZ5DbVfT4aVU1VfYxgnvixK5kpoEpi\\",\\"wpkh\\":\\"tb1qjqz2el2le7zqj7jsns4ndcmqfnkgwux3f5vrkl\\"}"`;

exports[`[Address][Util] getAddresses > Test-net > Should return correct test-net addresses for key2 1`] = `"{\\"tr\\":\\"tb1p2l8c6wap8c93726006v44sdmmv566mvrez7483pda0ylrg6fnaast6vncn\\",\\"pkh\\":\\"n2ziWGyRjz9gbrHX2x7di4MtZSXiefQ4nr\\",\\"wpkh\\":\\"tb1qawdvw9jjf6jql4y7gtv8pfrq3zhmemx9jlng4s\\"}"`;
